src/plugins/dashboard/server/routes/create.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { RequestHandlerContext } from '@kbn/core-http-request-handler-context-server';
+import type { VersionedRouter } from '@kbn/core-http-server';
+import type { ContentManagementServerSetup } from '@kbn/content-management-plugin/server';
+import { CONTENT_ID } from '@kbn/visualizations-plugin/common/content_management';
+import type { Dashboard } from '../../common/api/2023_10_31';
+import { v2023_10_31 } from '../../common/api';
+
+export function registerCreate(
+  router: VersionedRouter<RequestHandlerContext>,
+  cm: ContentManagementServerSetup
+) {
+  router
+    .post({
+      path: '/api/dashboard',
+      access: 'public',
+      description: 'Create a dashboard',
+    })
+    .addVersion(
+      {
+        version: '2023-10-31',
+        validate: {
+          request: {
+            body: v2023_10_31.dashboardCreate,
+          },
+          response: {
+            200: { body: v2023_10_31.dashboard },
+          },
+        },
+      },
+      async (ctx, req, res) => {
+        const client = cm.contentClient
+          .getForRequest({ request: req, requestHandlerContext: ctx })
+          .for<Dashboard>(CONTENT_ID);
+        let result: Dashboard;
+        try {
+          const cmResult = await client.create(req.body);
+          result = cmResult.result as unknown as Dashboard;
```

Comment:
Seems like there is a gap in the `Dashboard` type being passed through so I was getting `object`.

Reply:
cc @vadimkibana @sebelga

Reply:
To be investigated, in the code it _looks like_ we are forwarding the type in `CreateItemResponse<T>`

<img width="990" alt="Screenshot 2024-05-07 at 11 33 17" src="https://github.com/elastic/kibana/assets/2854616/1eb7dc19-6955-4aff-a707-03f43107cc71">

Reply:
I think the problem is `data: object` does not let the return type infer the type of the generic

Reply:
But here we are talking about the response right? You might be right but it is strange the the input type affects the output type. Although I've seen weirdest things with TS 😄